### PR TITLE
Fix: do not display BNPL methods and PMME when WooPayments is disabled

### LINKED ISSUE #9390: BNPL payment methods and PMME are still enabled when WooPayments gateway is disabled
### To Reproduce
On shortcode, disable WooPayments gateway in WooPayments Settings
Shop and observe that BNPL payment methods and PMME are enabled throughout the store

### Expected behavior
No payment methods are enabled on the front-end if the gateway is disabled (plugin still activated)

image

image

image

### PR DESCRIPTION
**Pull request by danielmx-dev**
Fixes #9390

#### Changes proposed in this Pull Request

When the main gateway is disabled, we expect all the WooPayments related to not be shown to the shoppers, including PMME and BNPL payment methods.

#### Testing instructions

* Disable the WooPayments gateway in the WooCommerce Settings
* Enable BNPL methods in WooPayments Settings **WITHOUT** enabling WooPayments itself.

#### Product and cart pages
**Current Behavior**: PMME are rendered
image

**Expected Behavior**: PMME are not displayed.

#### Shortcode checkout
**Current Behavior**: BNPL methods are enabled and available to choose from.
image

**Expected Behavior**: BNPL methods should not be displayed.

#### Regression Testing

* Enable the main WooPayments gateway and verify that PMME in product and cart. pages are rendered.
* In both shortcode and blocks checkout, verify that the BNPL methods are displayed and can be used to place an order.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9467` or branch name `fix/9390-hide-bnpl-pmme-if-gateway-is-disabled` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/9390-hide-bnpl-pmme-if-gateway-is-disabled) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: c35a242d92a5282946e05cbe1fb6abd98d6dfaa1
- Build time: 2024-09-24 12:57:13 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by danielmx-dev** [comment]
Since the change is relatively simple but affects a significant area of the functionality, I ran the E2E tests before merging to make sure we got a green build

https://github.com/Automattic/woocommerce-payments/actions/runs/10946463792

File: includes/class-wc-payments.php [hunk 1]
```diff
@@ -1813,6 +1813,9 @@ public static function init_woopay() {
 	 * @return void
 	 */
 	public static function load_stripe_bnpl_site_messaging() {
+		if ( ! self::get_gateway()->is_enabled() ) {
```
Comment (frosso) [inline includes/class-wc-payments.php]: (minor) For this, I was wondering if we should instead adding this check a few lines above, before the hooks are registered, with the other check: https://github.com/Automattic/woocommerce-payments/blob/525d198c5e99bf9bc85b5f5d7d2f0bd916eba12e/includes/class-wc-payments.php#L621

What do you think?

File: includes/class-wc-payment-gateway-wcpay.php [hunk 2]
```diff
@@ -825,6 +825,9 @@ public function needs_https_setup() {
 	 * @return bool Whether the gateway is enabled and ready to accept payments.
 	 */
 	public function is_available() {
+		if ( ! WC_Payments::get_gateway()->is_enabled() ) {
```
Comment (frosso) [inline includes/class-wc-payment-gateway-wcpay.php:828]: This makes sense, but I still wonder why it wasn't part of the codebase prior 😅 I have been searching past changes, but couldn't find an answer in the ancient scriptures.

**Review by frosso — APPROVED** [review]
:shipit: just a very minor question/comment

File: includes/class-wc-payments.php (hunk 1 above)
Comment (danielmx-dev) [reply includes/class-wc-payments.php]: Good idea! I implemented it in https://github.com/Automattic/woocommerce-payments/pull/9467/commits/b6c7d059875f9d1029d97270950d2d7d939f2ce8

**Review by frosso — APPROVED** [review]
Still looking good, thanks for the changes!

**Pull request merged**